### PR TITLE
Close #183 - Support Scala 3.0.0-M2

### LINKED ISSUE #183: Support Scala 3.0.0-M2
## Support Scala 3.0.0-M2
It looks like [this problem](https://github.com/hedgehogqa/scala-hedgehog/runs/1402680817?check_suite_focus=true#step:7:453) is solved in Scala 3.0.0-M2. Because of this, there was no doc file generated for 0.6.0 and 0.6.0 could not be published to the Maven Central since Maven Central doesn't accept any artifacts without the doc file.

So now Hedgehog supporting Dotty (Scala 3) can be published. It will be done once the new minitest supporting Scala 3.0.0-M2 is released (https://github.com/monix/minitest/pull/72).

### PR DESCRIPTION
**Pull request by Kevin-Lee**
Close #183 - Support Scala 3.0.0-M2

### CODE REVIEW

**Review by charleso — APPROVED** [review]

**Comment by dwijnand** [comment]
I see no reason to keep building and publish for 3.0.0-M1.  Was that kept intentionally?

**Comment by Kevin-Lee** [comment]
@dwijnand I have only one reason. We failed to release one for 3.0.0-M1 before and there might be some projects can't easily upgrade Scala from 3.0.0-M1 to 3.0.0-M2 due to breaking changes in 3.0.0-M2. 
It's just for this release so next one will not have all of them but the whatever is the latest. 
Do you think we should avoid 3.0.0-M1? I don't really have any preference here and need a second opinion.

**Comment by dwijnand** [comment]
Projects are just going to have to upgrade from M1 to M2, so it's pretty pointless IMO.  But 2 lines of extra YAML aren't that much of a hassle either :D

**Comment by Kevin-Lee** [comment]
@dwijnand Alright. Let's do it for just M2 then. Thank you!

File: build.sbt [hunk 1]
```diff
@@ -119,7 +115,11 @@ lazy val minitest = crossProject(JVMPlatform, JSPlatform)
         case Some((2L, 11L)) =>
           Seq("io.monix" %%% "minitest" % "2.8.2")
         case _ =>
-          Seq("io.monix" %%% "minitest" % "2.9.0")
+          val binaryScalaVersion = CrossVersion.binaryScalaVersion(scalaVersion.value)
+          if (binaryScalaVersion == "3.0.0-M1")
```
Comment (dwijnand) [inline build.sbt:119]: Trim

Comment (Kevin-Lee) [reply build.sbt]: Thanks!

**Review by dwijnand — APPROVED** [review]

**Pull request merged**